### ISSUE TITLE
Show outline for selected objects

### ISSUE THREAD

**Issue by inDigiNeous** [vizorvr/vizor#1430]
Draw a outline/highlight for active selected objects. Like this:

http://jsfiddle.net/Eskel/g593q/5/

**Comment by inDigiNeous** [comment]
Testcase here: http://app.geokone.net/xyz/threejs/test_outline.html

**Comment by inDigiNeous** [comment]
In order to get proper outlines for objects that don't have connected geometry (like cubes), we should do something like this for the outline, dilate it by some percentage:

```javascript
dilateGeometry = function(geometry, length) {
	// gather vertexNormals from geometry.faces
	var vertexNormals	= new Array(geometry.vertices.length);

	geometry.faces.forEach(function(face){
		if( face instanceof THREE.Face4 ){
			vertexNormals[face.a]	= face.vertexNormals[0];
			vertexNormals[face.b]	= face.vertexNormals[1];
			vertexNormals[face.c]	= face.vertexNormals[2];
			vertexNormals[face.d]	= face.vertexNormals[3];		
		}else if( face instanceof THREE.Face3 ){
			vertexNormals[face.a]	= face.vertexNormals[0];
			vertexNormals[face.b]	= face.vertexNormals[1];
			vertexNormals[face.c]	= face.vertexNormals[2];
		}else	console.assert(false);
	});

	// modify the vertices according to vertextNormal
	geometry.vertices.forEach(function(vertex, idx){
		var vertexNormal = vertexNormals[idx];
		vertex.x	+= vertexNormal.x * length;
		vertex.y	+= vertexNormal.y * length;
		vertex.z	+= vertexNormal.z * length;
	});		
};
```

**Comment by inDigiNeous** [comment]
Getting pretty frustrated with this. For some reason the masking pass just doesn't work. In the testcase here: git@github.com:inDigiNeous/threejs.git in the case test_outline2.html it works perfectly, but when this same code is applied inVizor, the mask just stays black.

Things i've checked now:
 * Are the composer passes and flags absolutely the same as in the test case ? Yes
 * Is the mask and outline rendering properly ? Yes
 * Do we even have a stencil buffer in our rendering context ? Yes, three.js creates this by default
 * Tested with different flags for the autoclear and so on for the passes, no luck

Currently I have no idea why the mask part is not cleared after the last phase runs

```javascript
		// Create our effect composer
		this.composer = new THREE.EffectComposer(this.renderer)

		// Need the stencils for the outline
		this.composer.renderTarget1.stencilBuffer = true
		this.composer.renderTarget2.stencilBuffer = true

		// Add the passes to our composer
		//
		// First we render the scene normally
		this.composer.addPass(this.renderPasses.scene)

		// Then we render the mask, that equals to the outlined object, but rendered only in full 0x00 color
		this.composer.addPass(this.renderPasses.mask)

		// Now we render the outline
		// We have the mask pass enabled, so we are not rendering over the masked area
		this.composer.addPass(this.renderPasses.outline)

		// Disable the stencil test 
		//
		// clearMask pass only does this;
		// var context = renderer.context;
		// context.disable( context.STENCIL_TEST );

		this.composer.addPass(this.renderPasses.clearMask)

		// Then we copy the resulting renderTarget onto the canvas
		this.composer.addPass(this.renderPasses.copy)
```

Everything looks fine, but for some reason the mask remains in the final copy phase, although the stencil test against the stencil buffer should be set in renderPasses.mask phase, and then renderPasses.outline should not write to the color buffer on the parts where the stencil buffer contains the value 0x000000.

Hmm. The only thing that comes to my mind is now that something else is rendering to the stencil buffer ? How would I verify that ..

Maybe should try out with different stencil buffer colors and masks.. let's see if that helps.

**Comment by inDigiNeous** [comment]
So this is where the selection is currently in vizor.
The mask pass is the black region, which I cannot get to work properly.

<img width="289" alt="foo" src="https://cloud.githubusercontent.com/assets/1212726/13876472/3e47baba-ed0d-11e5-8460-838b0c4f0c48.png">

**Comment by inDigiNeous** [comment]
Okay. I tried to get this working as far as possible, but I cannot find the reason without going into visually rendering the target context and seeing what is going on. So I pushed all the changes to this branch. 
My best guess is that for some reason something is not sized correctly and therefore the render contexts and the mask does not match.

See the testcase in my github repo for how it should work.
